### ISSUE TITLE
Now getting error 420

### ISSUE THREAD

**Issue by jessusmach** [mdesilva/AmazonFlexUnlimited#159]
![image](https://github.com/mdesilva/AmazonFlexUnlimited/assets/37351138/446aea5b-43eb-49d8-85e1-0250b264c8ee)

**Comment by leiyio** [comment]
Me too

**Comment by asadbek064** [comment]
app version: 2.123.1  

Error: Too many requests in a short period of time.

RAW log of the request:

` {'Server': 'Server', 'Date': 'Fri, 03 Nov 2023 17:07:13 GMT', 'Content-Type': 'application/json', 'Content-Length': '33', 'Connection': 'keep-alive', 'x-amz-rid': 'AYX82AFFRZVVWRN8WAKV', 'x-amzn-ErrorType': 'OperationFailedException:http://internal.amazon.com/coral/com.amazon.omwbuseyservice.offers/', 'x-amzn-RequestId': 'fb2af217-952f-48d4-9aff-7d88e2850aa4', 'Vary': 'Content-Type,Accept-Encoding,User-Agent', 'Strict-Transport-Security': 'max-age=47474747; includeSubDomains; preload'} <urllib3.response.HTTPResponse object at 0x7f78a0ee7550>`

**Comment by nobby-styles** [comment]
it's not captcha, because I'm getting it on all my request offers on Amazon flex, but not in the app. there's  no captcha

**Comment by nobby-styles** [comment]
Maybe the app request has changed slightly since an update. I've tried differing the delay to quite long like 4 s but it makes no diference

**Comment by nobby-styles** [comment]
There's discussion on the problem at the bottom of this thread. It's going to be very difficult to solve https://github.com/mdesilva/AmazonFlexUnlimited/issues/144

**Comment by vtdeleond** [comment]
Isn't this same problem with Check-in endpoint?

**Comment by Sam11172678** [comment]
It seams that this error is because of sending too many request in a short period of time, but I changed sending request speed (as slow as a human refreshing the page), and I still got the same error, but when you try to get an offer on the app you can easily do it even when refreshing the screen very fast, so I think it is because of how the code introduce itself to Amazon Server or API, there is couple of lines of code that is used for introducing the code, if it is an android or IOS device, using this version of IOS, etc. I think this part of code should be changed, and introduce the code in another way!

If anybody knows how to change it please advice!

**Comment by nobby-styles** [comment]
it's the new header value Signature-Input

**Comment by nemo03** [comment]
> it's the new header value Signature-Input

You are right, Amazon check if the request sent from Amazon Flex genuine app.

**Comment by jessusmach** [comment]
> it's the new header value Signature-Input
@nobby-styles 
How we can fix it?

**Comment by nobby-styles** [comment]
No one has a solution as of yet, requires knowing the crypto algorithm on the phone app. There's some progress on that but not the complete solution.

**Comment by asadbek064** [comment]
the only way I got it to work is if you run the app and Python code in the same VM box with a random delay, make different calls randomly between get offers, offload the tokens, and make sure the vmbox and your real app runs on the same network (proxy) and never use both at the same time etc. Imo this is not worth it I'm tired of updating the script every month. My method is very stupidly inefficient but it works, if anyone else has a different method please share
